EthermintEVM (#216)
subquery/documentation: docs/build/cosmos-evm.md

# Cosmos Ethermint (EVM) Support

We provide a custom data source processor for [Cosmos's Ethermint EVM](https://github.com/cosmos/ethermint). This offers a simple way to filter and index both EVM and Cosmos activity on many Cosmos networks within a single SubQuery project.

> **You can also refer to the basic [Cronos EVM](https://github.com/subquery/tutorials-frontier-evm-starter/tree/moonriver) example project with an event and call handler.**

Reviewer: Should this be a different link? 

I just found this on Naveens github https://github.com/DeveloperInProgress/ethermint-evm-starter

## Getting started

1. Add the custom datasource as a dependency:

- Create a new project from an EVM template though `subql init` OR

> - For existing projects, `yarn add @subql/ethermint-evm-processor` or `yarn add @subql/ethermint-evm-processor`.

Reviewer: ```suggestion
- For existing projects, `npm i @subql/ethermint-evm-processor` or `yarn add @subql/ethermint-evm-processor`.
```

2. Add a custom data source as described below.
3. Add handlers for the custom data source to your code.

## Data Source Spec

| Field             | Type                                                            | Required | Description                                |
| ----------------- | --------------------------------------------------------------- | -------- | ------------------------------------------ |

> | processor.file    | `'./node_modules/@subql/ethermint-evm-processor/dist/index.js'` | Yes      | File reference to the data processor code  |

Reviewer: ```suggestion
| processor.file    | `'./dist/ethermintEvm.js'` | Yes      | File reference to the data processor code  |
```

They will also need an exports like so https://github.com/subquery/acala-evm-starter/blob/main/package.json#L32

But `"ethermintEVM": "./node_modules/@subql/ethermint-evm-processor/dist/index.js"`

This is so IPFS deployments work.

## Event Handlers

| Field  | Type                           | Required | Description                                  |
| ------ | ------------------------------ | -------- | -------------------------------------------- |
| kind   | `substrate/EthermintEvmEvent`  | Yes      | Specifies that this is an Event type handler |
| filter | [Event Filter](#event-filters) | No       | Filter the data source to execute            |

### Event Filters

| Field  | Type         | Example(s)                                                        | Description                                                                                                                                      |
| ------ | ------------ | ----------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------ |
| topics | String array | `Transfer(address indexed from,address indexed to,uint256 value)` | The topics filter follows the Ethereum JSON-PRC log filters, more documentation can be found [here](https://docs.ethers.io/v5/concepts/events/). |

> <b>Note on topics:</b>

Reviewer 2: ```suggestion
**Note on topics:**
```
